sys/net/gnrc/network_layer/ipv6/static_addr/gnrc_ipv6_static_addr.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2022 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+
+#include "net/gnrc/ipv6/nib.h"
+#include "net/gnrc/ipv6.h"
+#include "net/gnrc/netapi.h"
+#include "net/gnrc/netif.h"
+#include "net/gnrc/rpl.h"
+#include "net/ipv6/addr.h"
+#include "net/netdev.h"
+#include "net/netopt.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+/**
+ * @brief   ID of the upstream interface, set to 0 to attempt auto-select
+ */
+#ifndef CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM
+#define CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM 0
+#endif
+
+/**
+ * @brief   ID of the downstream interface, set to 0 to attempt auto-select
+ */
+#ifndef CONFIG_GNRC_IPV6_STATIC_ADDR_DOWNSTREAM
+#define CONFIG_GNRC_IPV6_STATIC_ADDR_DOWNSTREAM 0
+#endif
+
+static void _config_upstream(gnrc_netif_t *upstream)
+{
+    ipv6_addr_t addr;
+
+    const char *static_addr =
+#ifdef GNRC_IPV6_STATIC_ADDR
+        GNRC_IPV6_STATIC_ADDR;
+#else
+        NULL;
+#endif
+
+    const char *static_route =
+#ifdef GNRC_IPV6_STATIC_DEFAULT_ROUTER
+        GNRC_IPV6_STATIC_DEFAULT_ROUTER;
+#else
+        NULL;
+#endif
+
+    DEBUG("gnrc_ipv6_static_addr: interface %u selected as upstream\n", upstream->pid);
+
+    /* configure static address */
+    int addr_len;
+    if (static_addr != NULL &&
+        (addr_len = ipv6_prefix_from_str(&addr, static_addr)) > 0) {
+        gnrc_netif_ipv6_addr_add_internal(upstream, &addr, addr_len,
+                                          GNRC_NETIF_IPV6_ADDRS_FLAGS_STATE_VALID);
+    }
+
+    /* configure static route */
+    if (static_route != NULL &&
+        ipv6_addr_from_str(&addr, static_route) != NULL) {
+        gnrc_ipv6_nib_ft_add(&addr, 0, &addr, upstream->pid, 0);
+    }
+}
+
+static void _config_downstream(gnrc_netif_t *downstream)
+{
+    const char *static_prefix =
+#ifdef GNRC_IPV6_STATIC_PREFIX
+        GNRC_IPV6_STATIC_PREFIX;
+#else
+        NULL;
+#endif
+
+    DEBUG("gnrc_ipv6_static_addr: interface %u selected as downstream\n", downstream->pid);
+
+    if (GNRC_IPV6_STATIC_PREFIX == NULL) {
+        return;
+    }
+
+    ipv6_addr_t prefix;
+    int len = ipv6_prefix_from_str(&prefix, static_prefix);
+    if (len <= 0) {
+        return;
+    }
+
+    /* configure subnet on downstream interface */
+    int idx = gnrc_netif_ipv6_add_prefix(downstream, &prefix, len,
+                                         UINT32_MAX, UINT32_MAX);
+    if (idx < 0) {
+        DEBUG("gnrc_ipv6_static_addr: adding prefix to %u failed\n", downstream->pid);
+        return;
+    }
+
+    /* start advertising subnet */
+    gnrc_ipv6_nib_change_rtr_adv_iface(downstream, true);
+
+    /* configure RPL root if applicable */
+    gnrc_rpl_configure_root(downstream, &downstream->ipv6.addrs[idx]);
```

Comment:
Mhhh this makes it at least a RPL route and as such not just any router. Maybe guard then with border router here?

Reply:
Actually this is just copied from `dhcpv6_client_conf_prefix()`/`uhcp_handle_prefix()` - `gnrc_rpl_configure_root()` turns into a no-op if `gnrc_rpl` is not used.

Reply:
> Actually this is just copied from `dhcpv6_client_conf_prefix()`/`uhcp_handle_prefix()` -

Yes, but those are used to configure a border router or at least the RPL root which, yes, might not be the border router. In https://github.com/RIOT-OS/RIOT/pull/18477#discussion_r962356422 you stated that this new module is not just intended for border routers. However, having multiple RPL roots in a network would lead to multiple overlapping DODAGs, so there needs to be some handling for when this is not supposed to be a RPL root, but a RPL router still, i.e., `gnrc_rpl` is used.